src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Services/SyntaxFacts/CSharpSyntaxFacts.cs

```diff
@@ -2132,5 +2132,38 @@ public bool IsThrowStatement(SyntaxNode node)
 
         public bool IsLocalFunction(SyntaxNode node)
             => node.IsKind(SyntaxKind.LocalFunctionStatement);
+
+        public SyntaxNode GetIOperationRootNode(SyntaxNode syntaxNode)
+        {
+            Contract.ThrowIfNull(syntaxNode);
+
+            var rootNode = syntaxNode
+                .AncestorsAndSelf(ascendOutOfTrivia: false)
+                .Where(IsRootOperationNode)
+                .LastOrDefault();
+
+            return rootNode;
+
+            static bool IsRootOperationNode(SyntaxNode node)
+                => (node is StatementSyntax && StatementIsRoot(node)) ||
```

Comment:
the `is statementsyntax` is redundant since you check if it's a block.

Reply:
Not true, ClassDeclaration and CompilationUnit are both block but not StatementSyntax

Reply:
oh.  i misread what StatementIsRoot does.  

note though that by doing `node is StatementSyntax` the only cae you support is `BlockSyntax` in statementIsroot. so this can definitely br broken up to be clearer.  i.e. haveing an explicit check for Compilationunit.  

(aside: i don't know why you'd need to care about ClassDeclarationSytnax (which is a *not* a block)).

Reply:
Removed in latest iteration